distros/spring/src/main/java/com/connexta/transformation/service/TransformController.java

```diff
@@ -13,86 +13,86 @@
+   * handles: - Validating the message - Forwarding the request to the service lookup service.
+   * Returns an HTTP Status Code of 201 Created on success and a location URI; otherwise, an error
+   * status code.
+   *
+   * @param acceptVersion - The API version used by the client to produce the REST message
+   * @param transformRequest - A request to transform a file into discovery metadata and other
+   *     supporting products
+   * @throws URISyntaxException if a string could not be parsed as a URI reference
+   * @throws TransformationException if an error occurs while executing this method
+   * @return response of the request and the URI for polling the status is returned in the Location
+   *     header of the response.
+   */
+  @RequestMapping(
+      value = "/transform",
+      consumes = {MediaType.APPLICATION_JSON_VALUE},
+      method = RequestMethod.POST)
   public ResponseEntity<Void> transform(
-      @ApiParam(
-              value =
-                  "The API version used by the client to produce the REST message. The client must accept responses marked with any minor versions after this one. This implies that all future minor versions of the message are backward compatible with all previous minor versions of the same message. ",
-              required = true)
-          @RequestHeader(value = "Accept-Version", required = true)
-          String acceptVersion,
-      @ApiParam(
-              value =
-                  "A request to transform a file into discovery metadata and other supporting products.",
-              required = true)
-          @Valid
-          @RequestBody
-          TransformRequest transformRequest) {
+      @RequestHeader(ACCEPT_VERSION) String acceptVersion,
+      @Valid @RequestBody TransformRequest transformRequest)
+      throws URISyntaxException, TransformationException {
+
     LOGGER.debug("{}: {}", ACCEPT_VERSION, acceptVersion);
-    forwardRequest(transformRequest);
-    return ResponseEntity.accepted().build();
+    return postURI(transformRequest);
   }
 
-  private void forwardRequest(TransformRequest transformRequest) {
-    LOGGER.debug("Forwarding request to: {}", lookupServiceUrl);
-    ResponseEntity<String> response =
-        restTemplate.postForEntity(lookupServiceUrl, transformRequest, String.class);
-    LOGGER.debug(
-        "HTTP response status code from lookup service: {}", response.getStatusCodeValue());
+  private ResponseEntity<Void> postURI(TransformRequest transformRequest)
+      throws TransformationException, URISyntaxException {
+
+    ResponseEntity<Void> response;
+    Transformation trans =
+        transformationManager.createTransform(
+            transformRequest.getCurrentLocation(),
+            transformRequest.getFinalLocation(),
+            transformRequest.getMetacardLocation());
+
+    URI s = ServletUriComponentsBuilder.fromCurrentRequest().build().toUri();
```

Comment:
We should probably create a test to make sure that this utilizes the "forwarded" header. I know that would mean we are essentially testing spring but I think it's important as a way to document and provide evidence for how the URL is being made in this method.